### PR TITLE
Rework MapLayer logic

### PR DESCRIPTION
**Pull request by bevans2000**
This PR focuses on refactoring the MapLayer logic in the Navigator Window as part of #1385. It addresses the following high level changes:
- Creation of the concept of a Surface Point of Interest based at a Coordinate that is well aligned to entities rendered on a Map layer
- A ```SurfaceManager``` class that provides a means to store ```SurfacePOI``` and provide efficient  geometric based searching
- A common ```SurfaceFeatureMapLayer``` to consolidate logic on selecting and rendering ```SurfacePOI```

The above are applied to Landmarks and Explore Sites.

A further optimization is to significantly reducing the searching required for tracking objects on the map. This is done by the map layers reporting what hotspots are valid. This allows a single small list to be maintained of only the visible hotspots.  
In addition, the hotspots now supports a tooltip capability that is available to Landmarks & Explored Sites.

There are various clean-ups on ```MapLayer```.

### CODE REVIEW

File: mars-sim-core/src/main/java/com/mars_sim/core/structure/Settlement.java [hunk 1]
```diff
@@ -917,54 +914,6 @@ public boolean timePassing(ClockPulse pulse) {
 		return true;
 	}
 	
-	/**
-	 * Checks and reports on the average mineral site improvement made.
-	 * 
-	 * @param sol
-	 */
-	private void reportsMineralSiteImprovement(int sol) {
```
Comment (mokun) [inline mars-sim-core/src/main/java/com/mars_sim/core/structure/Settlement.java:925]: Can we still have this method for reporting mineral site improvement at the end of a sol ?

File: mars-sim-core/src/main/resources/messages.properties [hunk 2]
```diff
@@ -914,6 +912,18 @@ NaturalAttributeType.teaching           = Teaching
 
 NavButtonDisplay.log.mediaTrackerError = NavButtonDisplay Media Tracker Error {0}
 
+LandmarkType.AO = Artificial Object
```
Comment (mokun) [inline mars-sim-core/src/main/resources/messages.properties:915]: It'll be great that one day we offer players the option to turn on and off each of these landmark types, wouldn't it ?

Comment (bevans2000) [reply mars-sim-core/src/main/resources/messages.properties:915]: Yes I agree. I want it that each layer can support filters. I'm working on ideas for it

File: mars-sim-core/src/main/java/com/mars_sim/core/structure/Settlement.java (hunk 1 above)
Comment (bevans2000) [reply mars-sim-core/src/main/java/com/mars_sim/core/structure/Settlement.java:925]: It is an expensive method and the output just goes to the console. If you want to record the historical improvements of sites then lets do it probably and uses our Metrics Manager idea

**Comment by mokun** [comment]
It seems after this PR or previous commits, the sun data panel no longer gets updated.

image

**Comment by bevans2000** [comment]
I doubt it is this PR

**Pull request merged**